### PR TITLE
ROX-23616,ROX-23618: bump cosign

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

Bumping cosign dep to the latest released version.

The version bump for x/sync broke a few of the delegated scanning unit tests, needed to update those within this PR as well.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

- see CI passing.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by janisz — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -119,20 +119,20 @@ require (
...
 	golang.org/x/text v0.14.0
 	golang.org/x/time v0.5.0
 	golang.org/x/tools v0.19.0
 	golang.stackrox.io/grpc-http1 v0.3.9
-	google.golang.org/api v0.157.0
-	google.golang.org/genproto v0.0.0-20240116215550-a9fa1716bcac
-	google.golang.org/grpc v1.61.1
+	google.golang.org/api v0.172.0
+	google.golang.org/genproto v0.0.0-20240311173647-c811ad7063a7
+	google.golang.org/grpc v1.62.1
```
Comment (janisz) [inline go.mod:135]: I think this breaks our test. How about adding a replace and pin it to 1.61.1?

**Review by dcaravel — CHANGES_REQUESTED** [review]
The version bump for `x/sync` broke a few of the delegated scanning unit tests, will need to update the tests.

Created a PR to demo a potential fix (based off this PR's branch): https://github.com/stackrox/stackrox/pull/10793

**Review by dcaravel — APPROVED** [review]
LGTM, thanks!

**Pull request merged**